cmd/operator-sdk/bundle/validate.go

```diff
@@ -72,19 +72,38 @@ To build and validate an image:
 
 `,
 		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			jsonOutput := newOutPut()
 			if err = c.validate(args); err != nil {
-				return fmt.Errorf("error validating args: %v", err)
+				msg := "Error validating args"
+				if c.json {
+					jsonOutput.addError(fmt.Errorf(msg))
+					jsonOutput.addError(err)
+					fmt.Printf("\nResult:\n%s\n", jsonOutput.toJSON())
```

Comment:
When `--json` is set, a command should never print non-JSON output. I suggest making a helper like the following for situations like this:
```go
func fatalJSON(msg string) {
	out := Output{
		{
			Type: errorType,
			Message: msg,
		},
	}
	fmt.Print(out.toJSON())
	os.Exit(1)
}
```
Then use like:
```go
if err = c.validate(args); err != nil {
	if c.json {
		fatalJSON(fmt.Sprintf("invalid command args: %v", err))
	}
	return fmt.Errorf("invalid command args: %v", err)
}
```